apisx/v1alpha1/xlistenerset_types.go

```diff
@@ -129,7 +129,10 @@ type ListenerEntry struct {
 	// same port, subject to the Listener compatibility rules.
 	//
 	// +optional
-	Port PortNumber `json:"port,omitempty"`
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	Port int32 `json:"port,omitempty"`
```

Comment:
note: I stopped using `PortNumber` because I couldn't get kubebuilder to do what I wanted with type aliases.

Though rob was able to do it here - https://github.com/kubernetes-sigs/gateway-api/pull/3748

So I'm wondering if my local CLIs are out of date maybe? 🤔

Reply:
What was happening with aliases? For me I noticed that it was ANDing validation with the type definition where I think this used to override validation, was that the issue you were running into?

Reply:
validations aren't showing up in the CRD when you have `type X = B` and I add validations to my usages of `type x`